src/NServiceBus.Gateway/Receiving/SingleCallChannelReceiver.cs

```diff
@@ -93,13 +108,32 @@ async Task HandleSubmit(CallInfo callInfo)
                 await stream.ReadAsync(body, 0, body.Length).ConfigureAwait(false);
                 args.Body = body;
 
-                if (await deduplicator.DeduplicateMessage(callInfo.ClientId, DateTime.UtcNow, new ContextBag()).ConfigureAwait(false))
+                var context = new ContextBag();
+                if (await deduplicationStorage.IsDuplicate(callInfo.ClientId, context).ConfigureAwait(false))
                 {
-                    await messageReceivedHandler(args).ConfigureAwait(false);
+                    Logger.InfoFormat("Message with id: {0} is already on the bus, dropping the request", callInfo.ClientId);
                 }
                 else
                 {
-                    Logger.InfoFormat("Message with id: {0} is already on the bus, dropping the request", callInfo.ClientId);
+                    await messageReceivedHandler(args).ConfigureAwait(false);
+                    try
+                    {
+                        await deduplicationStorage.MarkAsDispatched(callInfo.ClientId, context).ConfigureAwait(false);
+                    }
+                    catch (Exception e)
+                    {
+                        // swallow exception in non-dtc modes.
+                        // When using no transactions, the message has been sent to the transport already. Throwing would cause the operation to be retried and a guaranteed duplicate to be created. By swallowing the exception, the duplicate is only created if the same message is sent to the gateway for another reason.
+                        // When using distributed transactions, throw so that both persistence and transport can rollback atomically.
+                        if (useTransactionScope)
+                        {
+                            throw;
+                        }
+                        else
+                        {
+                            Logger.Warn($"Failed to mark message with id '{callInfo.ClientId}' as dispatched. This message won't be deduplicated.'", e);
```

Comment:
Can we add some more details as to why the message couldn't be dispatched? 

>.. since the transport and persister combination doesn't support distributed transactions

or something similar?

Reply:
the exception will hold more information about the issue. The message has actually been dispatched. Given that you can't use DTC with the combination in use, I'm not sure about the added value of mentioning DTC?